src/Kdyby/Events/EventManager.php

```diff
@@ -259,6 +262,30 @@ protected function extractSubscriber(EventSubscriber $unsubscribe)
 
 
 
+	/**
+	 * @param callable $unsubscribe
+	 * @return array
+	 */
+	protected function extractCallable(callable $unsubscribe)
+	{
+		$subscriber = $unsubscribe;
+		$unsubscribe = array();
+
+		foreach ($this->listeners as $event => $prioritized) {
```

Comment:
I don't see much sense in optimizing event remove use-cases. They're mostly for tests. But I'm not gonna try to prevent you from optimizing it :)